extension/index.googleFunction.js

```diff
@@ -7,31 +7,57 @@ const logger = utils.getLogger()
 
 exports.extensionTrigger = async (request, response) => {
   try {
+    const obj = request?.body?.resource?.obj
+    if (!obj) {
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 400,
+        body: {
+          errors: [
+            {
+              code: 'BadRequest',
+              message: 'Invalid body payload.',
+            },
+          ],
+        },
+      })
+    }
     const authToken = auth.getAuthorizationRequestHeader(request)
-    const { obj } = request.body.resource
     const paymentResult = await paymentHandler.handlePayment(obj, authToken)
     if (paymentResult.success) {
-      response.status(200).send({
-        actions: paymentResult.data ? paymentResult.data.actions : [],
-      })
-    } else {
-      response.status(400).send({
-        errors: paymentResult.data ? paymentResult.data.errors : undefined,
+      return utils.sendGoogleFunctionResponse({
+        response,
+        statusCode: 200,
+        body: {
+          actions: paymentResult.data ? paymentResult.data.actions : [],
+        },
       })
     }
+    return utils.sendGoogleFunctionResponse({
+      response,
+      statusCode: 400,
+      body: {
+        errors: paymentResult.data ? paymentResult.data.errors : undefined,
```

Comment:
I am curious why `errors` are nested within data? Why existence of data implies those are errors? Looks that there might be also update actions instead of errors: https://github.com/commercetools/commercetools-adyen-integration/blob/3d752e6dbf5a62febe288082b801a256ff07d3fa/extension/src/paymentHandler/payment-handler.js#L58

 Would be simpler if response object would look like that:
```
{
  "actions": [] // defined in case there are any
  "errors": [] // defined in case there are any
}
```

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/630/commits/2149fca06f83665f83c7bb39a36feedc99bb10d9

Reply:
@butenkor just realized that the above change would fail e2e test all the time. Shall we bypass it at this moment?

Reply:
We have it covered in the follow-up ticket.